### PR TITLE
work correctly info analysis args

### PR DESCRIPTION
**Pull request by hyuraku**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----
To soIve https://github.com/Homebrew/brew/issues/6677, I add the check `args.days` and `args.category` for `brew info --analytics` to `handle extra args`.

### CODE REVIEW

**Comment by EricFromCanada** [comment]
If `brew info` is going to accept `--category` and `days` for formula analytics, then `output_formula_analytics` should be checking that their values are valid just like `output_analytics` currently does. It should also be checking for and displaying the categories with hyphens instead of underscores (i.e. "install-on-request" and "build-error") for consistency.

**Comment by hyuraku** [comment]
@EricFromCanada 
> If brew info is going to accept --category and --days for formula analytics, then output_formula_analytics should be checking that their values are valid just like output_analytics currently does.

I understood. I make `output_formula_analytics` check whether `args.days` includes`valid_days` like `output_analytics`

> It should also be checking for and displaying the categories with hyphens instead of underscores

You mean that like below code?

```
> brew info solr --analytics --category="install-on-request"
1     | solr                                                                                                                                                    | 9,645 |  100.00%
==> install_on_request (30 days)
Index | Name (with options)                                                                                                                                     | Count |  Percent
-----:|-----|------:|--------:
1     | solr    |   710 |  100.00%
==> install_on_request (90 days)
Index | Name (with options)                                                                                                                                     | Count |  Percent
-----:|------|------:|--------:
1     | solr     | 1,821 |  100.00%
==> install_on_request (365 days)
Index | Name (with options)                                                                                                                                     | Count |  Percent
-----:|------|------:|--------:
1     | solr      | 8,673 |  100.00%
```

File: Library/Homebrew/cmd/info.rb [hunk 1]
```diff
@@ -357,6 +357,8 @@ def output_analytics(filter: nil)
   end
 
   def output_formula_analytics(f)
+    valid_days = %w[30 90 365]
+    valid_categories = %w[install install-on-request build-error]
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/info.rb]: Move these into a function also called by `output_analytics`

File: Library/Homebrew/cmd/info.rb [hunk 2]
```diff
@@ -369,6 +371,18 @@ def output_formula_analytics(f)
       value.each do |days, results|
         days = days.to_i
         if full_analytics
+          if args.days.present?
+            raise UsageError, "day must be one of #{valid_days.join(", ")}" unless valid_days.include?(args.days)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/info.rb]: Move these into a function also called by `output_analytics`

File: Library/Homebrew/cmd/info.rb [hunk 3]
```diff
@@ -369,6 +371,18 @@ def output_formula_analytics(f)
...
+            next if args.days&.to_i != days
+          end
+
+          if args.category.present?
+            unless valid_categories.include?(args.category)
+              raise UsageError, "category must be one of #{valid_categories.join(", ")}"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cmd/info.rb]: Move these into a function also called by `output_analytics`

**Comment by MikeMcQuaid** [comment]
Thanks so much for your first contribution! Without people like you submitting PRs we couldn't run this project. You rock, @hyuraku!

**Comment by hyuraku** [comment]
Thank you @MikeMcQuaid .

**Pull request merged**